### PR TITLE
Settings: Add postgres storage handler

### PR DESCRIPTION
**Pull request by fanatov37**

### CODE REVIEW

File: docker/php.ini [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+memory_limit = 536870912M
+
+xdebug.remote_enable=1
+xdebug.remote_handler=dbgp
+xdebug.remote_port=9000
+xdebug.remote_autostart=1
+xdebug.remote_connect_back=0
+xdebug.idekey=docker
+;IMPORTANT!!! need use your currently host. V
+xdebug.remote_host=10.254.254.254
```
Comment (JRoy) [inline docker/php.ini]: Does this need to be hardcoded/included?

File: docker/Dockerfile [hunk 2]
```diff
@@ -29,9 +41,4 @@ RUN wget https://johnvansickle.com/ffmpeg/releases/ffmpeg-release-amd64-static.t
 RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/bin/ --filename=composer
 
 # Set the WORKDIR to /app so all following commands run in /app
-WORKDIR /var/www/html
-
-COPY . ./
-
-# Install dependencies with Composer.
-RUN composer install --prefer-source --no-interaction
```
Comment (JRoy) [inline docker/Dockerfile]: Why were theses removed

File: src/Settings/Storage/Components/PDOStorage.php [hunk 3]
```diff
@@ -267,7 +267,7 @@ public function openUser(
 
         // Retrieve and cache the existing user data row if available.
         try {
-            $sth = $this->_pdo->prepare("SELECT id, settings, cookies FROM `{$this->_dbTableName}` WHERE (username=:username)");
+            $sth = $this->_pdo->prepare("SELECT id, settings, cookies FROM {$this->_dbTableName} WHERE (username=:username)");
```
Comment (JRoy) [inline src/Settings/Storage/Components/PDOStorage.php]: Why was this changed?

File: src/Settings/Storage/Components/PDOStorage.php [hunk 4]
```diff
@@ -247,7 +247,7 @@ public function deleteUser(
     {
         try {
             // Just attempt to delete the row. Doesn't error if already missing.
-            $sth = $this->_pdo->prepare("DELETE FROM `{$this->_dbTableName}` WHERE (username=:username)");
+            $sth = $this->_pdo->prepare("DELETE FROM {$this->_dbTableName} WHERE (username=:username)");
```
Comment (JRoy) [inline src/Settings/Storage/Components/PDOStorage.php]: Why was this changed?

File: src/Settings/Storage/Components/PDOStorage.php [hunk 5]
```diff
@@ -227,7 +227,7 @@ public function moveUser(
             }
 
             // Now attempt to rename the old username column to the new name.
-            $sth = $this->_pdo->prepare("UPDATE `{$this->_dbTableName}` SET username=:newusername WHERE (username=:oldusername)");
+            $sth = $this->_pdo->prepare("UPDATE {$this->_dbTableName} SET username=:newusername WHERE (username=:oldusername)");
```
Comment (JRoy) [inline src/Settings/Storage/Components/PDOStorage.php]: Why was this changed?

File: src/Settings/Storage/Components/PDOStorage.php [hunk 6]
```diff
@@ -192,7 +192,7 @@ public function hasUser(
         $username)
     {
         // Check whether a row exists for that username.
-        $sth = $this->_pdo->prepare("SELECT EXISTS(SELECT 1 FROM `{$this->_dbTableName}` WHERE (username=:username))");
+        $sth = $this->_pdo->prepare("SELECT EXISTS(SELECT 1 FROM {$this->_dbTableName} WHERE (username=:username))");
```
Comment (JRoy) [inline src/Settings/Storage/Components/PDOStorage.php]: Why was this changed?

File: src/Settings/Storage/Components/PDOStorage.php [hunk 7]
```diff
@@ -159,10 +159,10 @@ protected function _setUserColumn(
             // Update if the user row already exists, otherwise insert.
             $binds = [':data' => $data];
             if ($this->_cache['id'] !== null) {
-                $sql = "UPDATE `{$this->_dbTableName}` SET {$column}=:data WHERE (id=:id)";
+                $sql = "UPDATE {$this->_dbTableName} SET {$column}=:data WHERE (id=:id)";
                 $binds[':id'] = $this->_cache['id'];
             } else {
-                $sql = "INSERT INTO `{$this->_dbTableName}` (username, {$column}) VALUES (:username, :data)";
+                $sql = "INSERT INTO {$this->_dbTableName} (username, {$column}) VALUES (:username, :data)";
```
Comment (JRoy) [inline src/Settings/Storage/Components/PDOStorage.php]: Why was this changed?

File: src/Settings/Storage/Components/PDOStorage.php [hunk 8]
```diff
@@ -159,10 +159,10 @@ protected function _setUserColumn(
             // Update if the user row already exists, otherwise insert.
             $binds = [':data' => $data];
             if ($this->_cache['id'] !== null) {
-                $sql = "UPDATE `{$this->_dbTableName}` SET {$column}=:data WHERE (id=:id)";
+                $sql = "UPDATE {$this->_dbTableName} SET {$column}=:data WHERE (id=:id)";
```
Comment (JRoy) [inline src/Settings/Storage/Components/PDOStorage.php]: Why was this changed?

File: src/Settings/Storage/Postgresql.php [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+
+namespace InstagramAPI\Settings\Storage;
+
+use InstagramAPI\Settings\Storage\Components\PDOStorage;
+use PDO;
+
+/**
+ * Class Postgresql.
+ */
+class Postgresql extends PDOStorage
```
Comment (JRoy) [inline src/Settings/Storage/Postgresql.php]: Should be `PostgreSQL`

File: src/Settings/Storage/Postgresql.php [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+{
+    /**
+     * Postgresql constructor.
+     *
+     * {@inheritdoc}
+     */
+    public function __construct()
+    {
+        parent::__construct('Postgresql');
```
Comment (JRoy) [inline src/Settings/Storage/Postgresql.php]: Should be `PostgreSQL`

File: src/Settings/Storage/Components/PDOStorage.php (hunk 3 above)
Comment (fanatov37) [reply src/Settings/Storage/Components/PDOStorage.php]: bc it will be error for postgres.
  **`** is wrong syntax for postgres

http://joxi.ru/5mdoggJU3WVpgr

File: src/Settings/Storage/Components/PDOStorage.php (hunk 4 above)
Comment (fanatov37) [reply src/Settings/Storage/Components/PDOStorage.php]: bc it will be error for postgres.

File: src/Settings/Storage/Components/PDOStorage.php (hunk 3 above)
Comment (fanatov37) [reply src/Settings/Storage/Components/PDOStorage.php]: for mysql and sqlite is OK without **`**

File: src/Settings/Storage/Postgresql.php (hunk 9 above)
Comment (fanatov37) [reply src/Settings/Storage/Postgresql.php]: ok. gonna to rename this class

File: src/Settings/Storage/Postgresql.php (hunk 10 above)
Comment (fanatov37) [reply src/Settings/Storage/Postgresql.php]: ok +

File: docker/php.ini (hunk 1 above)
Comment (fanatov37) [reply docker/php.ini]: So I use mac  and this remote host for xdebug will be normal. 
Just need do alis like this 
`sudo ifconfig lo0 alias 10.254.254.254`

for ubuntu alias did not work and my collegues use dockerbridge ip.
cmd `ifconfig`. 

So I really don't know 1 way for mac and ubuntu xdebug settings **remote_host**

File: docker/Dockerfile (hunk 2 above)
Comment (fanatov37) [reply docker/Dockerfile]: **COPY** unnecessary bc we use docker-compose **volumes**  

**composer install** use after **docker-compose up -d**

File: composer.json [hunk 11]
```diff
@@ -47,6 +47,8 @@
         "ext-gd": "*",
         "ext-exif": "*",
         "ext-zlib": "*",
+        "ext-pdo": "*",
+        "ext-redis": "*",
```
Comment (JRoy) [inline composer.json]: Please move these to the `suggest` section of the `composer.json` as they are not needed 100% of the time

File: docker/Dockerfile [hunk 12]
```diff
@@ -19,7 +19,19 @@ RUN echo "deb http://deb.debian.org/debian jessie main" > /etc/apt/sources.list
     && docker-php-ext-configure gd --with-freetype-dir=/usr/include/ --with-jpeg-dir=/usr/include/ \
```
Comment (JRoy) [inline docker/Dockerfile]: Please keep the Dockerfile in the root directory of the repository

Comment (fanatov37) [reply docker/Dockerfile]: It's not problem I can move.

but we will use just **docker-compose**  in root directory
`docker-compose up -d`

it's better flow for project than run single DockerFile :)

So your choose ?

Comment (JRoy) [reply docker/Dockerfile]: I just don't want to break people's existing workflows with docker so if you could move it back that would be best :P

Comment (fanatov37) [reply docker/Dockerfile]: ok. I understand.
 no problem. 

I think in the future we will use only docker-compose :)

File: composer.json (hunk 11 above)
Comment (fanatov37) [reply composer.json]: +

File: docker/Dockerfile (hunk 12 above)
Comment (fanatov37) [reply docker/Dockerfile]: `+`

**Comment by JRoy** [comment]
Thanks for your contribution :)
